javaslang/src/test/java/javaslang/collection/euler/Utils.java

```diff
@@ -49,7 +49,10 @@ public static BigInteger factorial(int n) {
     public static boolean isPrime(long val) {
         return API.Match(val).of(
                 Case($(2L), true),
-                Case($(n -> n > 2), n -> !primes().takeWhile(d -> d <= Math.sqrt(n)).exists(d -> n % d == 0)),
+                Case($(n -> n > 2), n -> {
```

Comment:
if you're interested for values `<2`, I think those should come before the `$(2L)`